### ISSUE TITLE
Multiple Coast Lines

### ISSUE THREAD

**Issue by kheaactua** [kheaactua/diplomacy-engine#4]
The machanism I thought would limit fleets from jumping coasts, say from ocean to ocean through Panama, will actually limit fleet movements in an unacceptable way.

I thought of a superior way, but it will probably not be implemented in time.

What I propose is that important coast lines be entered into the data.  Anywhere territory with two coasts, should have those coast lines defined.

They would be defined as the structure:

``` json
[
   {
      "coast_id": 2,
      "coast-lines": [
         ["water_territory_id", "land_territory_id"],
         // ...
      ]
   }
]
```

(this would be saved in two DB tables)

Then, I'd be able to see if a feet was trying to jump from one coast to the next.

Currently, without this information, there is no way for me to tell what coast they are on.. I can check they're source, but if they've moved up a coast line for a few turns, I won't know where they're from.

For the spreadsheet, this would require a sheet be made, using the territory IDs in the sheet showing all the coasts.  I'd suggest a sheet with an array of pair columns,

``` text
A          B         C             D       E  
Water    Land                    Water   Land
1          2        <blank>       23      34 
2          5        <blank>       24      45 
```

Where the IDs are territory IDs.

Again, this only has to be done for multi-coastline territories (e.g. Panama, Spain, ..)

Then I could parse these into a structure and load them into the DB.

**Comment by gobyrne** [comment]
Yeah, I realized last night that even saving the last water territory a boat was in doesn't really help since you can move all the way around a peninsula using the coasts and then go to sea on the other side.

Let's not worry about coding this for now. We can leave this to the players for the time being.

**Comment by kheaactua** [comment]
Yeah, that's the plan.  I tags this with the 1.1 release milestone, meaning after this weekend.

**Comment by kheaactua** [comment]
(Just jotting down an idea)

The comment above about defining coast lines (either programmatically or by input data) still requires saving unit movements history (which becomes complex when supplying/disbanding)

One alternate idea would be implementing sub-territories.  i.e. Panama would be divided into three sub-territories.  This way the parent territory, Panama, would still be occupied, but it's sub categories would simulate coastlines.

The would also impact the data provided from the spreadsheet.  Basically every sub-territory would need to know it's adjacencies.  There are only a couple territories that fall under this problem though.

... again (if any one else is reading) just brainstoriming
